### ISSUE TITLE
Persistent data structures

### ISSUE THREAD

**Issue by ShalokShalom** [zero-functional/zero-functional#62]
Hi :)

I consider Nim as a language and as such, I hope to find some way to keep my data persistent. 

Does this project evaluate that and if thats a no, can you recommend one?

**Comment by michael72** [comment]
Hi @ShalokShalom ,

can you rephrase your question and make it a real issue - cause you opened which normally is for having a specific problem - e.g. I got the code or idea - how do I do that with this library?
Also `persistent data` is probably confusing: do you mean persistent data storage (which actually is the more common use of the word _persistent_) or immutable data structures? I guess the latter?
zero_functional is "just" a way to use functional programming algorithms with the nim programming language. It does not aim at providing a complete environment for functional programming - so basically immutable data structures are supported as far as they can be iterated. The main use cases all create new objects leaving the traversed data untouched - exception is (the only) meta function with side-effects: that is `foreach`. See documentation.

**Comment by ShalokShalom** [comment]
https://en.wikipedia.org/wiki/Persistent_data_structure
